src/selection-util.js

```diff
@@ -23,24 +23,42 @@ export const isRowSelected = (rowIndex, selection) => {
   return rowIndex <= y2 && rowIndex >= y1
 }
 
-export const left = (selection, expand) => {
+export const left = (selection, expand, colCount, rowCount) => {
   const { x1, y1, x2, y2 } = selection
 
   if (!expand) {
-    const x = Math.max(x1 - 1, 0)
-    return { x1: x, x2: x, y1, y2: y1 }
+    let x = x1 - 1
+    let y = y1
+    if (x < 0) {
+      x = colCount - 1
+      if (y > 0) {
+        y--
+      } else {
+        y = rowCount - 1
+      }
+    }
+    return { x1: x, x2: x, y1: y, y2: y }
   }
   const x = Math.max(x2 - 1, 0)
 
   return { x1, x2: x, y1, y2 }
 }
 
-export const right = (selection, expand, colCount) => {
+export const right = (selection, expand, colCount, rowCount) => {
   const { x1, y1, x2, y2 } = selection
 
   if (!expand) {
-    const x = Math.min(x1 + 1, colCount - 1)
-    return { x1: x, x2: x, y1, y2: y1 }
+    let x = x1 + 1
+    let y = y1
+    if (x > colCount - 1) {
+      x = 0
+      if (y < rowCount - 1) {
```

Comment:
same as comment before. I believe the y and x values should be kept in bound independently

Reply:
on right overflow, need to move selection in only down direction, so when overflow occur, just need to check for y < maxRow